docs/installation/prepare/microcontrollers/_index.md

```diff
@@ -1,387 +1,351 @@
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+mkdir -p ~/esp
+cd ~/esp
+git clone --depth 1 -b v4.4.4 --single-branch --recurse-submodules --shallow-submodules https://github.com/npmenard/esp-idf
+```
+
+Then, install the required tools for ESP-IDF:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+cd ~/esp/esp-idf
+./install.sh esp32
+```
+
+To activate ESP-IDF, run the following command to source (`.`) the activation script `export.sh`:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+. $HOME/esp/esp-idf/export.sh
+```
+
+To avoid conflicts with other toolchains, adding this command to your `.bashrc` or `.zshrc` is not recommended.
+Save this command to run in any future terminal session where you need to activate the ESP-IDF development framework.
+
+#### Install the Rust ESP Toolchain and Activate the ESP-RS Virtual Environment
+
+First, install the following tools with `cargo`:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+cargo install espup
+```
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+cargo install cargo-espflash v2.0.0-rc.1
+```
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+cargo install ldproxy
+```
+
+To download and install the esp-rs toolchain, run:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+espup install -s -f ~/esp/export-rs.sh -v 1.67.0
+```
+
+To activate the ESP Rust toolchain, run the following command to source (`.`) the activation script `export-rs.sh`:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+. $HOME/esp/esp-idf/export-rs.sh
+```
+
+To avoid conflicts with other toolchains, adding this command to your `.bashrc` or `.zshrc` is not recommended.
+Save this command to run in any future terminal session where you need to activate the ESP Rust development framework.
+
+## Install the Micro-RDK
+
+### Create a New Robot
+
+Navigate to [the Viam app](https://app.viam.com) and [add a new robot](/manage/fleet/robots/#add-a-new-robot) in your desired location.
+
+* Click on the name of the robot to go to the robot's page.
+* Skip the instructions in the [**Setup** tab](/manage/fleet/robots/#setup), as the setup instructions there are not for microcontrollers.
+* Keep your `Mode` and `Architecture` selections at default.
+
+### Configure an `esp32` Board
+
+{{< alert title="Info" color="info" >}}
+The`esp32` [board](/components/board/) model is not currently provided for you as an option in [the Viam app](https://app.viam.com), so you cannot use the **Config Builder** to configure this board.
+{{< /alert >}}
+
+To add an `esp32` board, navigate to the **Config** tab of your robot's page in [the Viam app](https://app.viam.com) and select **Raw JSON** mode.
+
+Copy the following JSON template and paste it into your configuration:
```

Comment:
```suggestion
Copy the following JSON template.
Paste into your configuration inside of the`"components": []` object:
```

Reply:
@mattjperez Do you think this would be sufficient to direct people? I can also include what's going to be autogenerated in the JSON, if you think that would be better!

Reply:
I think a complete JSON file they can drop into the Raw JSON field will be the best approach for now as even how to add an `esp32` board component via the builder isn't obvious since it requires typing into a drop-down box.

Keeping the original wording and using this json would be the most straight forward I think.
```json
{
  "components": [
    {
      "attributes": {
        "pins": [
          15
        ],
        "analogs": [
          {
            "name": "soil",
            "pin": "34"
          }
        ]
      },
      "depends_on": [],
      "name": "board",
      "type": "board",
      "model": "esp32"
    }
  ]
}
```

Reply:
I disagree. Complete Raw JSON should be avoided at all costs by users. I don't see anything in this config that would warrant Raw JSON.
Typing the the model "esp32" and populating just the attributes in the normal config builder is a pretty standard flow for advanced configs. That's the modular components experience for example.

Reply:
Should the instructions be made into two steps then?
1) add `esp32` board using builder 
2) add `{ "pins": [15] }` to the new board's attributes using the new board's menu

Reply:
Yes

Reply:
Spoke with @npentrel offline. She pointed out that we have raw json for all other components on the docs site. I don't like it, but I think consistency is more important.